tests/components/matrix/test_send_message.py

```diff
@@ -1,41 +1,121 @@
+                ATTR_DATA: {ATTR_FORMAT: FORMAT_TEXT},
+                ATTR_TARGET: list(TEST_JOINABLE_ROOMS.keys()),
+            },
+            {
+                "target_rooms": list(TEST_JOINABLE_ROOMS.keys()),
+                "message_type": "m.room.message",
+                "content": {"msgtype": "m.text", "body": "Test text message"},
+            },
+        ),
+        (
+            "HTML message",
+            {
+                ATTR_MESSAGE: "Test <b>html</b> message",
+                ATTR_DATA: {ATTR_FORMAT: FORMAT_HTML},
+                ATTR_TARGET: list(TEST_JOINABLE_ROOMS.keys()),
+            },
+            {
+                "target_rooms": list(TEST_JOINABLE_ROOMS.keys()),
+                "message_type": "m.room.message",
+                "content": {
+                    "msgtype": "m.text",
+                    "body": "Test <b>html</b> message",
+                    "format": "org.matrix.custom.html",
+                    "formatted_body": "Test <b>html</b> message",
+                },
+            },
+        ),
+        (
+            "Bot (notice) message",
+            {
+                ATTR_MESSAGE: "Test bot (notice) message",
+                ATTR_DATA: {ATTR_FORMAT: FORMAT_NOTICE},
+                ATTR_TARGET: list(TEST_JOINABLE_ROOMS.keys()),
+            },
+            {
+                "target_rooms": list(TEST_JOINABLE_ROOMS.keys()),
+                "message_type": "m.room.message",
+                "content": {
+                    "msgtype": "m.notice",
+                    "body": "Test bot (notice) message",
+                },
+            },
+        ),
+    ],
+)
 async def test_send_message(
-    hass: HomeAssistant, matrix_bot: MatrixBot, image_path, matrix_events, caplog
+    hass: HomeAssistant,
+    matrix_bot: MatrixBot,
+    matrix_events: list,
+    caplog: pytest.LogCaptureFixture,
+    ids: str,
+    data: dict,
+    expected_attributes: dict,
 ):
     """Test the send_message service."""
 
     await hass.async_start()
     assert len(matrix_events) == 0
     await matrix_bot._login()
 
-    # Send a message without an attached image.
-    data = {ATTR_MESSAGE: "Test message", ATTR_TARGET: list(TEST_JOINABLE_ROOMS)}
     await hass.services.async_call(
         MATRIX_DOMAIN, SERVICE_SEND_MESSAGE, data, blocking=True
     )
 
+    matrix_bot._handle_multi_room_send.assert_called_once_with(**expected_attributes)
+
     for room_alias_or_id in TEST_JOINABLE_ROOMS:
         assert f"Message delivered to room '{room_alias_or_id}'" in caplog.messages
 
-    # Send an HTML message without an attached image.
+
+async def test_send_image(
+    hass: HomeAssistant,
+    matrix_bot: MatrixBot,
+    image_path: io.BytesIO,
+    matrix_events: list,
+    caplog: pytest.LogCaptureFixture,
+):
+    """Test send a message with an attached image."""
+
+    await hass.async_start()
+    assert len(matrix_events) == 0
+    await matrix_bot._login()
```

Comment:
Does each test need these lines? If yes, can we make a fixture or common function for it?